lib/dataset/bins.cpp

```diff
@@ -219,8 +219,10 @@ Variable lookup_previous(const DataArray &function, const Variable &x, Dim dim,
 Variable pretend_bins_for_threading(const DataArray &da, Dim bin_dim) {
   const auto dim = da.dims().inner();
   const auto size = std::max(scipp::index(1), da.dims()[dim]);
-  const auto nthread = size > 10000000  ? 24
-                       : size > 1000000 ? 4
+  const auto nthread = size > 8000000   ? 24
```

Comment:
Should these numbers really be hard-coded or should they depend on how many threads are available on the machine?

It could be that the benchmarks don't look as good on a laptop or a visa instance with less than 24 cores.

Reply:
It will not actually make that many threads. Instead, it cuts the input into chunks, which will be processed independently. Scipp's `transform` automatically uses multi-threading across these, so if you have fewer threads each thread will process multiple chunks. If chunks are too small there is extra cost to this, but I hope at this level it is not relevant.

Regarding thread counts in general, also see #3565 which would probably lead to more gains.